### PR TITLE
Polish language guide document

### LINKED ISSUE #2289: SELECT statement should be the same in TRAIN/PREDICT/EXPLAIN
- The `SELECT` clauses should be the same in TRAIN/PREDICT/EXPLAIN. For example, when there is `SELECT a * 2, b` in training, it should be also `SELECT a * 2, b` in predicting and explaining. We should add some docs to explain it.

- The column names of tables in TRAIN/PREDICT/EXPLAIN should be exactly the same. For example, the column names in train table is `a, b` and the column name in test table must be `a, b` too. Should we only need the equality of column number instead of column names? Moreover, can we use alias column names in PREDICT/EXPLAIN?

### LINKED ISSUE #2290: EVALUATE clause support
Do we support `TO EVALUATE` ? I see it in codes and unittests, but there is no doc on it.

https://github.com/sql-machine-learning/sqlflow/blob/94153c2b5c1133de7184b36df0663ad545d4ad09/pkg/parser/extended_syntax_parser.y#L257

https://github.com/sql-machine-learning/sqlflow/blob/32b5dc5cb5d34a4140aec7d4f4b329a3d55be27b/cmd/sqlflowserver/main_test.go#L1504

### PR DESCRIPTION
**Pull request by typhoonzero**
Fix https://github.com/sql-machine-learning/sqlflow/issues/2290
Fix https://github.com/sql-machine-learning/sqlflow/issues/2289

### CODE REVIEW

File: doc/language_guide.md [hunk 1]
```diff
@@ -328,6 +330,39 @@ WITH
...
+
+```sql
+TO EVALUATE model_table_reference
+[WITH
+  attr_expr [, attr_expr ...]]
+INTO evaluate_result_table;
+```
+
+- *model_table_reference* indicates that this model will be used to do the evaluation. The result metrics can show the performance of this model when using this model to predict "new" data.
+- *attr_expr* set attributes when doing evaluation. You can set `validation.metrics` to indicate which metrics will be outputed to the result table, e.g. `validation.metrics="Accuracy,AUC"`, you can find more supported metric names for Keras models [here](https://www.tensorflow.org/api_docs/python/tf/keras/metrics) and XGBoost models [here](https://xgboost.readthedocs.io/en/latest/parameter.html).
```
Comment (sneaxiy) [inline doc/language_guide.md]: `*attr_expr* sets attributes` ?

File: doc/language_guide.md [hunk 2]
```diff
@@ -328,6 +330,39 @@ WITH
...
+FROM table_references
+  [WHERE where_condition]
+  [LIMIT row_count]
+TO EVALUATE model_table_reference
+[WITH
+  attr_expr [, attr_expr ...]]
+INTO evaluate_result_table;
+```
+
+The select statement syntax is the same as the select statement syntax in the training syntax. When doing the evaluation, you should use a dataset that is not used for training to test evaluate the model dealing with "new" data. The evaluation result will be saved in `evaluate_result_table` when the evaluation job completes.
```
Comment (sneaxiy) [inline doc/language_guide.md]: `... that is not used for training to evaluate...`?

**Review by sneaxiy — COMMENTED** [review]
Almost LGTM for tiny changes.

**Review by sneaxiy — DISMISSED** [review]
LGTM.

File: doc/language_guide.md [hunk 3]
```diff
@@ -260,6 +260,8 @@ TO PREDICT result_table_reference
 USING model_table_reference;
 ```
 
+**NOTE: The select statement must have the same logic used in training, e.g. you use `SELECT a, b * 2 FROM train_data TO TRAIN` to train a model, when predicting, you must use also `SELECT a, b * 2 FROM pred_data TO PREDICT` so that the model will behave properly. You should do the same when using `TO EXPLAIN` and `TO EVALUATE` clause.**
```
Comment (wangkuiyi) [inline doc/language_guide.md]: - We don't need **NOTE** here.

----

Please be aware that the `SELECT ...` statement before the `TO PREDICT ..` clause must generate the same data schema as the one before the `TO TRAIN` clause does.  This requirement ensures that the data used in prediction has the same schema as those used in training.

File: doc/language_guide.md [hunk 4]
```diff
@@ -328,6 +330,39 @@ WITH
 USING TreeExplainer;
 ```
 
+## Evaluation Syntax
+
+A SQLFlow prediction statement consists of a sequence of select, predict, and using clauses.
```
Comment (wangkuiyi) [inline doc/language_guide.md]: ~~a sequence of~~

File: doc/language_guide.md [hunk 5]
```diff
@@ -328,6 +330,39 @@ WITH
...
+FROM table_references
+  [WHERE where_condition]
+  [LIMIT row_count]
+TO EVALUATE model_table_reference
+[WITH
+  attr_expr [, attr_expr ...]]
+INTO evaluate_result_table;
+```
+
+The select statement syntax is the same as the select statement syntax in the training syntax. When doing the evaluation, you should use a dataset that is not used for training to evaluate the model's performance when dealing with "new" data. The evaluation result will be saved in `evaluate_result_table` when the evaluation job completes.
```
Comment (wangkuiyi) [inline doc/language_guide.md]: The SELECT statement before `TO EVALUATE` must generate the same data schema as that used in training.  Please be aware that, by the machine learning theory, you are not supposed to use the training dataset for evaluation.  If you do that, you are likely to see a very good evaluation result, which indeed, doesn't tell much useful information.  Please write a SELECT statement as the prefix that generates different dataset but have the same schema.

The INTO clause names the table for saving evaluation results, which, in this example, is `evaluate_result_table`.

File: doc/language_guide.md [hunk 6]
```diff
@@ -328,6 +330,39 @@ WITH
...
+
+### Evaluate and Into Clause
+
+The *evaluate clause* describes which model will be used to evaluate, what metrics will be used for evaluation, and which table will be used to save the result.
```
Comment (wangkuiyi) [inline doc/language_guide.md]: *evaluate clause* => `TO EVALUATE` clause

describes which model => specifies the model

will be used to evaluate => to evaluate

**Review by wangkuiyi — COMMENTED** [review]
A few comments on English writing.

File: doc/language_guide.md (hunk 5 above)
Comment (typhoonzero) [reply doc/language_guide.md]: Thanks for the hints. Done and updated.

**Review by wangkuiyi — APPROVED** [review]

**Pull request merged**